### PR TITLE
Update test engine version to 19.03.12

### PR DESCRIPTION
**Pull request by thaJeztah**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -66,7 +66,7 @@ def runTests = { Map settings ->
     throw new Exception("Need test image object, e.g.: `runTests(testImage: img)`")
   }
   if (!dockerVersion) {
-    throw new Exception("Need Docker version to test, e.g.: `runTests(dockerVersion: '1.12.3')`")
+    throw new Exception("Need Docker version to test, e.g.: `runTests(dockerVersion: '19.03.11')`")
```
Comment (thaJeztah) [inline Jenkinsfile:69]: Not really important, but thought it would be clearer to use a more realistic example in the error

**Comment by thaJeztah** [comment]
Interesting; build failing on Jenkins;

https://ci-next.docker.com/public/blue/organizations/jenkins/docker-py/detail/PR-2584/1/pipeline

```
 docker build -t dockerbuildbot/docker-py:py2-5dea12f791bb8771553d21478015548bd7bb877f -f tests/Dockerfile --build-arg PYTHON_VERSION=2.7 .
...
Digest: sha256:cfa62318c459b1fde9e0841c619906d15ada5910d625176e24bf692cf8a2601d
Status: Downloaded newer image for python:2.7
 ---> 68e7be49c28c
Step 3/19 : ARG APT_MIRROR
 ---> Running in 686edd7e1190
Removing intermediate container 686edd7e1190
 ---> 733dc05c5fd9
Step 4/19 : RUN sed -ri "s/(httpredir|deb).debian.org/${APT_MIRROR:-deb.debian.org}/g" /etc/apt/sources.list  && sed -ri "s/(security).debian.org/${APT_MIRROR:-security.debian.org}/g" /etc/apt/sources.list
 ---> Running in 1969c958d278
cgroups: cgroup mountpoint does not exist: unknown
script returned exit code 1
```

This is on an Ubuntu 19.10 machine;

> Running on EC2 (managed-agents) - ubuntu-1910-overlay2 (i-01a0f5382e883adc8) in /home/ubuntu/workspace/docker-py_PR-2584

**Comment by thaJeztah** [comment]
Oh! I think these are the nodes that were setup for testing cgroups v2 @StefanScherer ?

File: Jenkinsfile [hunk 2]
```diff
@@ -31,7 +31,7 @@ def buildImages = { ->
 }
 
 def getDockerVersions = { ->
-  def dockerVersions = ["19.03.5"]
+  def dockerVersions = ["19.03.11"]
   wrappedNode(label: "ubuntu && !zfs && amd64") {
```
Comment (StefanScherer) [inline Jenkinsfile:35]: It picked Ubuntu 19.10 randomly.
Should we pin it better by using a label `ubuntu-1804`?

Comment (StefanScherer) [reply Jenkinsfile:35]: Or should we remove the ubuntu-1910 machine again? Maybe better and add ubuntu-2004 LTS machines instead. 🤔

Comment (thaJeztah) [reply Jenkinsfile:35]: Yes, probably, or do you have ideas for a labelling scheme where those nodes are not selected in the "regular" pools?

Comment (thaJeztah) [reply Jenkinsfile:35]: oh, our comments crossed LOL

I just opened https://github.com/docker/docker-py/pull/2585

**Comment by thaJeztah** [comment]
rebased, as https://github.com/docker/docker-py/pull/2585 was merged

@StefanScherer @ulyssessouza PTAL

**Comment by thaJeztah** [comment]
Updated to 19.03.12

**Pull request merged**